Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/Game.java

```diff
@@ -0,0 +1,115 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.util.ArrayList;
+
+public class Game {
+	protected 	Dictionary 	dictionary;		
+	protected 	Player 		player;		  	
+	protected 	char[] 		etalon;			
+	protected	ArrayList 	<Character> 	guessedLetters;
+	protected	char[] 		openedWord;		
+	protected	ArrayList	<Character>		wrongLetters;
+	protected 	UserInterface ui;			
+	protected 	final int 	TOTAL_GUESSES=9;
+	protected 	int 		currentGuesses;
+
+
+	public Game(){
+		dictionary = new ArrayDictionary ("words.txt");
+		ui = new GUIMainWindow(this);
+		player = new SimplePlayer(ui.playerName());
+		player.save();
+		ui.greet(player);
+		reset();
+	}
+
+	public void reset(){
+		etalon = dictionary.getRandomWord().toCharArray();
+		openedWord= new char[etalon.length];
+		for(int i=0;i<etalon.length;i++)
+			openedWord[i]='_';
+		currentGuesses = TOTAL_GUESSES;
+		guessedLetters =new ArrayList<Character>();	
+		wrongLetters =new ArrayList<Character>();
+	}
+
+	public String getOpened(){
+		return new String(openedWord);
+	}
+
+	public Player getPlayer(){
+		return player;
+	}
+
+	public ArrayList <Character> getGuessedLetters(){
+		return wrongLetters;
+	}
+
+
+	protected boolean hasWon(){
+		return etalon.equals(openedWord);
+	}
+
+	public int getHangmanStage(){
+		return TOTAL_GUESSES-currentGuesses;
+	}
+
+	public int attempt(char letter){
+		boolean found=false;
+		if(letter ==0){
+			return -1;
+		}
+		for(char l:guessedLetters){
+			if (l==letter){
+				return -1;
+			}
+		}
+		guessedLetters.add(letter);
+		for(int i=0; i<etalon.length;i++){
+			if(letter == etalon[i]){
+				found=true;
+				openedWord[i]=letter;
+			}
+		}
+
+		if(found){return 1;}
+		else{
+			currentGuesses--;
+			wrongLetters.add(letter);
+			return 0;
+		}
+	}
+
+	protected boolean gameOver(){
```

Comment:
I quite like the usage of "is" in methods which return booleans, e.g. this could be called `isGameOver` which would make what the method does more explicit.